crates/red_knot/src/main.rs

```diff
@@ -37,6 +38,17 @@ struct Args {
     )]
     current_directory: Option<SystemPathBuf>,
 
+    #[arg(
+        long,
+        help = "Path to the virtual environment the project uses",
+        long_help = "\
+Path to the virtual environment the project uses. \
+If provided, red-knot will use the `site-packages` directory of this virtual environment \
+to resolve type information for the project's third-party dependencies.",
+        value_name = "PATH"
+    )]
+    venv_path: Option<SystemPathBuf>,
```

Comment:
Along the lines of another comment below -- could we make this more general and just take the path to a Python binary to use? Virtual environments are not that "special" , and I don't think our CLI should make them seem more special than they are. We can easily support both virtual and non-virtual Python installations, and I think we should.

Reply:
One other reason why I called it `--venv-path` is that that's the name pyright gives this option

How would we name it if we wanted it to be more general? `--python` isn't quite right, because it's not the path to the Python installation that's being passed. `red_knot --venv-path .venv` doesn't point to a Python installation (`red_knot --venv-path .venv/bin/python` would, but that's more for users to type, and not all users will necessarily know where the Python installation is inside their venv directory)

Reply:
`--python-home` would be a name that matches CPython's `PYTHONHOME` env var, and CPython's naming in general, to refer to the directory containing `lib` / `Lib`.

I think it might make sense to have both `--venv-path` and `--python-home` as options (they might even just be aliases that do the same thing)? And it's fine with me if we just call it `--venv-path` for now and defer the system Pythons question for now; we aren't locking in any CLI choices yet.